frontend/src/components/smart-contracts/hooks/useJob.ts

```diff
@@ -47,6 +54,14 @@ const loadJobFromJobId = async (
   jobId: string,
   contracts: ISmartContractState
 ) => {
+  const storage = window.localStorage;
+  const jobCacheKey = `job:${jobId}`;
+  const cachedJobString = storage.getItem(jobCacheKey);
```

Comment:
Would there ever be a time where we would want to fetch a fresh copy? Probably not right, since our contracts are immutable..

Reply:
The blockchain and IPFS are immutable of course.  But we might need a fresh copy if we change how jobs are transformed after reading from the blockchain or IPFS.  I included a version key we can increment later to force a refresh for users.

Also, we will need to clear the localstorage cache often during development.